### ISSUE TITLE
PLR6201 fails but cannot be fixed on `list in tuple(list, list)`

### ISSUE THREAD

**Issue by gboeing** [astral-sh/ruff#9928]
Thanks for developing and supporting this wonderful tool! I'm using ruff 0.2.0. Say I have a bit of code to check if some list appears in a predefined collection of lists:

```python
def f(x):
    list1 = [1, 2, 3]
    list2 = [4, 5, 6]
    return x in (list1, list2)
```

Running `ruff . --preview` fails with "PLR6201 Use a `set` literal when testing for membership"... which you certainly should do whenever you can, but you cannot in this case. Python lists are not hashable, so you cannot insert them into a set. If you try, like `return x in {list1, list2}`, you get "TypeError: unhashable type: 'list'".

PLR6201 should not fail in this case because it is unfixable. Instead, PLR6201 should only fail if the items in the membership test are hashable.

**Comment by charliermarsh** [comment]
Agreed. It's hard to fix this _in general_ because we need arbitrary type inference, but we can do better in cases like the one above.

**Comment by gboeing** [comment]
Perhaps a simple type check for `collections.abc.Hashable` could resolve it in this specific instance at least.
